e2e/rbd.go

```diff
@@ -68,15 +81,112 @@ var _ = Describe("RBD", func() {
 			}
 
 			By("create a PVC and Bind it to an app", func() {
-				pvcPath := rbdExamplePath + "pvc.yaml"
-				appPath := rbdExamplePath + "pod.yaml"
 				validatePVCAndAppBinding(pvcPath, appPath, f)
 			})
 
 			By("create a PVC and Bind it to an app with normal user", func() {
-				pvcPath := rbdExamplePath + "pvc.yaml"
 				validateNormalUserPVCAccess(pvcPath, f)
 			})
+
+			By("create a PVC clone and Bind it to an app", func() {
+				createRBDSnapshotClass(f)
+				pvc, err := loadPVC(pvcPath)
+				if err != nil {
+					Fail(err.Error())
+				}
+
+				pvc.Namespace = f.UniqueName
+				framework.Logf("The PVC  template %+v", pvc)
+				err = createPVCAndvalidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				// validate created backend rbd images
+				images := listRBDImages(f)
+				if len(images) != 1 {
+					framework.Logf("backend image count %d expected image count %d", len(images), 1)
+					Fail("validate backend image failed")
+				}
+				snap := getSnapshot(snapshotPath)
+				snap.Namespace = f.UniqueName
+				snap.Spec.Source.Name = pvc.Name
+				snap.Spec.Source.Kind = "PersistentVolumeClaim"
+				err = createSnapshot(&snap, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				snapList, err := listSnapshots(f, "replicapool", images[0])
+				if err != nil {
+					Fail(err.Error())
+				}
+				if len(snapList) != 1 {
+					framework.Logf("backend snapshot not matching kube snap count,snap count = % kube snap count %d", len(snapList), 1)
+					Fail("validate backend snapshot failed")
+				}
+
+				validatePVCAndAppBinding(pvcClonePath, appClonePath, f)
+
+				err = deleteSnapshot(&snap, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+				err = deletePVCAndValidatePV(f.ClientSet, pvc, deployTimeout)
+				if err != nil {
+					Fail(err.Error())
+				}
+			})
+
+			// skipped raw pvc test in travis
+			// By("create a block type PVC and Bind it to an app", func() {
+			// 	validatePVCAndAppBinding(rawPvcPath, rawAppPath, f)
+			// })
+
+			By("create/delete multiple PVC and App", func() {
```

Comment:
@Madhu-1 what about making this function of creation more generic in util ? so that cephfs and rbd can just call that instead of repeating it.

Reply:
the functionality includes 3 different validation
* PVC create and bind it to app
* validate backend rbd images (doesnot apply for cephfs)
* delete app and PVC
IMO combining both create and delete operation is not a good option